### ISSUE TITLE
Do I need to use `spacemacs|add-company-backends` after `company` has been loaded?

### ISSUE THREAD

**Issue by ReimuXMX** [syl20bnr/spacemacs#9992]
Hello, and forgive my poor English...

I am an Emacs rookie, but now I need to create a layer.

So  do I need to use `spacemacs|add-company-backends` after `company` has been loaded (e.g.: in `(with-eval-after-load 'company)`)? 

Thanks.

**Comment by ReimuXMX** [comment]
Thank you for your reply!

In other words, if I want to migrate this macro to my own configuration, do I still have to put it after the `company` is loaded?

Thanks

**Comment by sdwolf** [comment]
I've never tried it myself... but i believe yes, if you want to use the macro without the spacemacs `post-init` hook you will need to have all the company modules loaded.

**Comment by ReimuXMX** [comment]
Thank you very much!

And...you're right, if you do not ensure that the `company` is loaded correctly, it may happen that the macro first executes. Which led to the local variable `company-backends` is void, I don't know the underlying principles.

I used `(with-eval-after-load 'company)` to avoid this error.

e.g.:
```emacs-lisp
(with-eval-after-load 'company
  (spacemacs|add-company-backends
   :backends company-irony
   :modes c-mode c++-mode objc-mode))
```

Since I did not use `use-package` for my simple configuration for root, I had to do this ...
